### PR TITLE
refactor(datepicker): move weekdays into table header

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

**Comment by fxck** [comment]
Is this the last thing needed before it gets merged to master?

**Comment by mmalerba** [comment]
@fxck It's getting close but I still want to replace `SimpleDate` with something that allows users to plug in [moment.js](https://momentjs.com) or other date implementations.

File: src/lib/datepicker/month-view.ts [hunk 1]
```diff
@@ -74,7 +74,13 @@ export class MdMonthView implements AfterContentInit {
   /** The date of the month that today falls on. Null if today is in another month. */
   _todayDate: number;
 
-  constructor(private _locale: CalendarLocale) {}
+  /** The names of the weekdays. */
+  _weekdays: string[];
+
+  constructor(private _locale: CalendarLocale) {
+    this._weekdays = this._locale.narrowDays.slice(this._locale.firstDayOfWeek)
```
Comment (jelbourn) [inline src/lib/datepicker/month-view.ts]: Add a comment like 
```ts
// Rotate the labels for days of the week based on the configured first day of the week.
```

File: src/lib/datepicker/calendar.scss [hunk 2]
```diff
@@ -103,3 +91,30 @@ $mat-calendar-next-icon-transform: translateX(-2px) rotate(45deg);
...
+  font-size: $mat-calendar-weekday-table-font-size;
+  font-weight: normal;
+  padding: 0 0 $mat-calendar-padding 0;
+}
+
+.mat-calendar-table-header-divider {
+  position: relative;
+  height: $mat-calendar-header-divider-width;
+
+  &::after {
```
Comment (jelbourn) [inline src/lib/datepicker/calendar.scss]: Add a comment explaining what the `::after` is used for?

**Pull request merged**